sdks/crypto-helper/src/payload.rs

```diff
@@ -0,0 +1,165 @@
+use std::{ffi::CString, os::raw::c_char};
+
+use espresso_types::{NsTable, Payload};
+use hotshot_types::traits::BlockPayload;
+
+#[repr(C)]
+pub struct Transaction {
+    // Transaction Namespace ID
+    namespace: u64,
+    // Transaction payload ptr
+    payload_ptr: *const u8,
+    // Transaction payload len
+    payload_len: usize,
+    // Transaction payload capacity (for freeing)
+    payload_cap: usize,
+}
+
+#[repr(C)]
+pub struct DecodingResult {
+    // The operation succeeded
+    pub success: bool,
+    // The error message if the operation failed, otherwise null
+    pub error: *mut c_char,
+    // Pointer to the array of transactions (null on error)
+    pub transactions: *mut Transaction,
+    // Number of transactions in the array
+    pub transactions_len: usize,
+    // Capacity of the transactions array (for freeing)
+    pub transactions_cap: usize,
+}
+
+impl DecodingResult {
+    fn err(msg: &str) -> DecodingResult {
+        let ptr = CString::new(msg).unwrap().into_raw();
```

Comment:
Will this `unwrap` cause any issue? Should we do something like `unwrap_or_else(|_| CString::new("error").unwrap())` or is it unnecessary?